src/Snackbar/Snackbar.js

```diff
@@ -20,15 +20,14 @@ function getStyles(props, context, state) {
   const styles = {
     root: {
       position: 'fixed',
-      left: 0,
+      left: '50%',
       display: 'flex',
-      right: 0,
       bottom: 0,
       zIndex: zIndex.snackbar,
       visibility: open ? 'visible' : 'hidden',
       transform: open ?
-        'translate(0, 0)' :
-        `translate(0, ${desktopSubheaderHeight}px)`,
+        'translate(-50%, 0)' :
+        `translate(-50%, ${desktopSubheaderHeight}px)`,
```

Comment:
I think that the following change would help in supporting different height. (hard-coded today)
```js
`translate(-50%, 100%)`,
```